src/uproot/source/http.py

```diff
@@ -607,7 +609,7 @@ def __repr__(self):
         if len(self._file_path) > 10:
             path = repr("..." + self._file_path[-10:])
         fallback = ""
-        if self._fallback is not None:
+        if hasattr(self, "_fallback") and self._fallback is not None:
```

Comment:
Just a little simpler:

```suggestion
        if getattr(self, "_fallback", None) is not None:
```

Reply:
It must have been getting here because it was raising an exception before `self._fallback` was even defined/added to the instance (somewhere in `__init__`, before line 576). Another solution would be to move the

```python
        self._fallback = None
```

earlier, but this is fine.